interfaces/repositories.go

```diff
@@ -0,0 +1,302 @@
+	defer row.Close()
+	return row.Next(), err
+}
+
+func NewDbLibraryRepo(dbHandlers map[string]DbHandler) *DbLibraryRepo {
+	dbLibraryRepo := new(DbLibraryRepo)
+	dbLibraryRepo.dbHandlers = dbHandlers
+	dbLibraryRepo.dbHandler = dbHandlers["DbLibraryRepo"]
+	return dbLibraryRepo
+}
+
+func (repo DbLibraryRepo) Store(library usecases.Library) error {
+	if !repo.libraryExisted(library.Id) {
+		_, err := repo.dbHandler.Execute(`INSERT INTO libraries (user_id) VALUES ($1)`,
+			library.User.Id)
+		if err != nil {
+			return err
+		}
+	}
+
+	for _, game := range library.Games {
+		_, err := repo.dbHandler.Execute(`INSERT INTO games (game_id, library_id)
+			VALUES ($1, $2)`, game.Id, library.Id)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (repo DbLibraryRepo) Remove(library usecases.Library) error {
+	_, err := repo.dbHandler.Execute(`DELETE FROM libraries WHERE id=$1`, library.Id)
+	return err
+}
+
+func (repo DbLibraryRepo) FindById(id int) (usecases.Library, error) {
+	row, err := repo.dbHandler.Query(`SELECT user_id FROM libraries WHERE id = $1 LIMIT 1`, id)
+	if err != nil {
+		return usecases.Library{}, err
+	}
+
+	var userId int
+	defer row.Close()
+	row.Next()
+	row.Scan(&userId)
+	userRepo := NewDbUserRepo(repo.dbHandlers)
+	user, err := userRepo.FindById(userId)
+	if err != nil {
+		library := usecases.Library{}
+		return library, err
+	}
+	library := usecases.Library{Id: id, User: user}
+
+	var gameId int
+	gameRepo := NewDbGameRepo(repo.dbHandlers)
+	row, err = repo.dbHandler.Query(`SELECT game_id FROM games WHERE library_id = $1`, library.Id)
+	if err != nil {
+		return library, err
+	}
+	defer row.Close()
+	for row.Next() {
+		err = row.Scan(&gameId)
+		if err != nil {
+			return library, err
+		}
+		game, err := gameRepo.FindById(gameId)
+		if err != nil {
+			return library, err
+		}
+		library.Games = append(library.Games, game)
+	}
+	return library, err
+}
+
+func (repo DbLibraryRepo) libraryExisted(id int) bool {
+	row, _ := repo.dbHandler.Query(`SELECT id FROM libraries
+		WHERE id=$1 LIMIT 1`, id)
+	defer row.Close()
+	return row.Next()
+}
+
+func (repo DbLibraryRepo) Count() (int, error) {
+	row, err := repo.dbHandler.Query(`SELECT COUNT(*) FROM libraries`)
+	if err != nil {
+		return 0, err
+	}
+	var count int
+	defer row.Close()
+	row.Next()
+	err = row.Scan(&count)
+	return count, err
+}
+
+func NewDbGameRepo(dbHandlers map[string]DbHandler) *DbGameRepo {
+	dbGameRepo := new(DbGameRepo)
+	dbGameRepo.dbHandlers = dbHandlers
+	dbGameRepo.dbHandler = dbHandlers["DbGameRepo"]
+	return dbGameRepo
+}
+
+func (repo DbGameRepo) Store(game usecases.Game) error {
+	_, err := repo.dbHandler.Execute(`INSERT INTO games (library_id, game_name, producer, value)
+    	VALUES ($1, $2, $3, $4)`, game.LibraryId, game.Name, game.Producer, game.Value)
+	return err
+}
+
+func (repo DbGameRepo) Remove(game usecases.Game) error {
+	_, err := repo.dbHandler.Execute(`DELETE FROM games WHERE game_id=$1`, game.Id+1)
```

Comment:
As I said before, all logic code (here is `game.Id +1`) should be done some elsewhere. Repositories just do its job - playing with database.